### PR TITLE
added pre persist method to queue listener and to implementations

### PR DESCRIPTION
**Pull request by orius123**
Signed-off-by: Eliya Sadan <eliyasadan@gmail.com>

### CODE REVIEW

**Comment by orius123** [comment]
@genadi-hp @oritstone please review.
@oritstone please merge if you feel like it.

File: engine/queue/score-queue-impl/src/main/java/org/openscore/engine/queue/services/ExecutionQueueServiceImpl.java [hunk 1]
```diff
@@ -73,6 +73,15 @@ public void enqueue(List<ExecutionMessage> messages) {
 				stateMessages.add(msg);
 			}
 		}
+
+        if (CollectionUtils.isNotEmpty(listeners)) {
+            long t = System.currentTimeMillis();
```
Comment (meirwah) [inline engine/queue/score-queue-impl/src/main/java/org/openscore/engine/queue/services/ExecutionQueueServiceImpl.java]: you know there is a normal StopWatch for that:
[https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/time/StopWatch.html]

File: engine/queue/score-queue-impl/src/main/java/org/openscore/engine/queue/services/ExecutionQueueServiceImpl.java [hunk 2]
```diff
@@ -73,6 +73,15 @@ public void enqueue(List<ExecutionMessage> messages) {
...
+            for (QueueListener listener : listeners) {
+                listener.prePersist(messages);
```
Comment (meirwah) [inline engine/queue/score-queue-impl/src/main/java/org/openscore/engine/queue/services/ExecutionQueueServiceImpl.java]: Are you running all listeners here? don't you need specific ones for the prePresist?

**Comment by meirwah** [comment]
:+1:

**Comment by orius123** [comment]
change to stop watch.
@meirwah when you are right you are right

**Comment by oritstone** [comment]
:+1:

**Pull request merged**